lib/cylc/network/httpclient.py

```diff
@@ -286,160 +311,146 @@ def call_server_impl(self, http_request_items):
+            if "timed out" in str(exc):
+                raise ClientTimeout(url, exc)
             else:
-                payload = None
-                json_headers = {'Content-Length': 0}
-            opener.addheaders = headers_list
-            req = urllib2.Request(url, payload, json_headers)
-
-            # This is an unpleasant monkey patch, but there isn't an
-            # alternative. urllib2 uses POST if there is a data payload
-            # but that is not the correct criterion.
-            # The difference is basically that POST changes
-            # server state and GET doesn't.
-            req.get_method = lambda: method
-            try:
-                response = opener.open(req, timeout=self.timeout)
-            except urllib2.URLError as exc:
-                if "unknown protocol" in str(exc) and url.startswith("https:"):
-                    # Server is using http rather than https, for some reason.
-                    sys.stderr.write(self.ERROR_NO_HTTPS_SUPPORT.format(exc))
-                    raise CylcError(
-                        "Cannot issue commands over unsecured http.")
-                if cylc.flags.debug:
-                    traceback.print_exc()
-                if "timed out" in str(exc):
-                    raise ClientTimeout(url, exc)
-                else:
-                    raise ClientConnectError(url, exc)
-            except Exception as exc:
-                if cylc.flags.debug:
-                    traceback.print_exc()
-                raise ClientError(url, exc)
-
-            if response.getcode() == 401:
-                access_desc = 'private'
-                if self.auth == self.ANON_AUTH:
-                    access_desc = 'public'
-                raise ClientDeniedError(url, self.prog_name, access_desc)
-            response_text = response.read()
-            if response.getcode() >= 400:
-                exception_text = get_exception_from_html(response_text)
-                if exception_text:
-                    sys.stderr.write(exception_text)
-                else:
-                    sys.stderr.write(response_text)
-                raise ClientConnectedError(
-                    url,
-                    "%s HTTP return code" % response.getcode())
-            if self.auth and self.auth[1] != NO_PASSPHRASE:
-                self.srv_files_mgr.cache_passphrase(
-                    self.suite, self.owner, self.host, self.auth[1])
+                raise ClientConnectError(url, exc)
+        except Exception as exc:
+            if cylc.flags.debug:
+                traceback.print_exc()
+            raise ClientError(url, exc)
+
+        if response.getcode() == 401:
+            access_desc = 'private'
+            if self.auth == self.ANON_AUTH:
+                access_desc = 'public'
+            raise ClientDeniedError(url, self.prog_name, access_desc)
+        response_text = response.read()
+        if response.getcode() >= 400:
```

Comment:
I think they are subtly different, with a few lines of duplication here. Just not enough to justify a re-factor. (I think it will be better if we simply declare that `requests` is an essential dependency and remove the logic for `urllib2`.)